addons/mail/static/src/models/message/message.js

```diff
@@ -106,17 +114,47 @@ function factory(dependencies) {
                     ];
                 }
             }
-            if ('channel_ids' in data && data.channel_ids) {
-                // AKU FIXME: side-effect of calling convert...
-                const channelList = [];
-                for (const channelId of data.channel_ids) {
-                    const channel = this.env.models['mail.thread'].insert({
-                        id: channelId,
-                        model: 'mail.channel',
-                    });
-                    channelList.push(channel);
+            if ('moderation_status' in data) {
+                data2.moderation_status = data.moderation_status;
+                const moderationMailbox = this.env.messaging.moderation;
+                const originThread =
+                    data.res_id &&
+                    data.model &&
+                    this.env.models['mail.thread'].find(
+                        t => t.id === data.res_id && t.model === data.model
+                    );
+
+                if (data.moderation_status === 'pending_moderation') {
+                    // Server will return the message without any channels as it's
+                    // in pending_moderation, so we need to add it to its moderation box
+                    // AND in the origin channel manually for the current partner if the message is
+                    // moderated by him
+                    if (moderationMailbox) {
+                        if (originThread && originThread.isModeratedByCurrentPartner) {
+                            data2.threadCaches.push(
+                                ['link', moderationMailbox.mainCache],
+                                ['link', originThread.mainCache],
+                            );
+                        }
+                    }
+                } else {
+                    if (moderationMailbox) {
+                        data2.threadCaches.push(['unlink', moderationMailbox.mainCache]);
+                    }
+                }
+                // Server will return the message without any channels as it's
+                // in pending_moderation, so we need to add it to its origin channel
+                // manually for the current partner if the message comes from him
+                // so he can see his messages that are "pending moderation"
+                if (
+                    originThread &&
+                    this.env.messaging.currentPartner &&
+                    'author_id' in data &&
+                    data.author_id &&
+                    this.env.messaging.currentPartner.id === data.author_id[0])
```

Comment:
This could be done generic for any partner. If they know they message in the store, they should see it.

It's a flux specific coincidence that only the author will receive it, it doesn't need to be enforced at this level.

Reply:
so whatever happen here we should insert ?

Reply:
This is ugly and weak, and the solution to cover all cases and have it nice is with rework of thread/messages links.

I'm against merging this PR before, because it makes code uglier and harder to changes (at least it will delay even more my PR). Also moderation is almost used by no-one, so I don't feel the urge to fix this specific moderation problem ASAP right now, especially when this is almost the lesser consequence of a much bigger problem that I'm currently fixing.

Reply:
> so whatever happen here we should insert ?

In the context of the current PR yes, the solution is to always insert message pending moderation in proper threadCache:

if current partner is moderator: in moderation box
in all other cases (including moderator): also in origin thread

Reply:
How would this delay your PR? If the new code is strong you should be able to just delete the diff here?

Reply:
> How would this delay your PR? If the new code is strong you should be able to just delete the diff here?

Expect maybe the tests, rest of diff of this PR becomes immediately obsolete by my PR. Gives extra effort to resolve conflict on rebase, other PRs will make changes based on this PR and may be ready for review before my PR again, etc.

I don't say that for complain, but I think it's stupid to wanting to merge this PR for the super small gain it gives, while delaying the much needed fixes. IMO the better move is to not merge this PR yet.

Reply:
Then, I'll stop refining it for now, we'll see in the future if it still needs to be